administrator/components/com_admin/script.php

```diff
@@ -8533,4 +8536,40 @@ protected function moveRemainingTemplateFiles()
 			}
 		}
 	}
+
+	/**
+	 * Ensure the core templates are correctly moved to the new mode.
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function fixTemplateMode()
+	{
+		$db = Factory::getDbo();
```

Comment:
see https://github.com/joomla/joomla-cms/pull/36511

```suggestion
		$db = Factory::getContainer()->get('DatabaseDriver');
```